packages/playwright-core/src/server/browserType.ts

```diff
@@ -92,23 +92,22 @@ export abstract class BrowserType extends SdkObject {
     return browser;
   }
 
-  async launchPersistentContext(metadata: CallMetadata, userDataDir: string, options: channels.BrowserTypeLaunchPersistentContextOptions & { useWebSocket?: boolean }): Promise<BrowserContext> {
-    options = this._validateLaunchOptions(options);
+  async launchPersistentContext(metadata: CallMetadata, userDataDir: string, persistentContextOptions: channels.BrowserTypeLaunchPersistentContextOptions & { useWebSocket?: boolean }): Promise<BrowserContext> {
+    const launchOptions = this._validateLaunchOptions(persistentContextOptions);
     if (this._useBidi)
-      options.useWebSocket = true;
+      launchOptions.useWebSocket = true;
     const controller = new ProgressController(metadata, this);
-    const persistent: channels.BrowserNewContextParams = { ...options };
```

Comment:
this was introduced [here](https://github.com/microsoft/playwright/commit/9569cb5c1e5c01c7d4f3246875d778fe1421c84d#diff-6a35d60cc44656b79b6202dac37dda6541caed378a6a83358b9329e3ebf07491R80), probably fine to remove it now since we don't modify it anymore (client certificates), alternatively we can make a copy at the beginning of the function call.